src/bin/deku_node.ml

```diff
@@ -591,18 +610,33 @@ let info_add_trusted_validator =
 let add_trusted_validator node_folder address =
   let open Network in
   let%await identity = read_identity ~node_folder in
+  let%await state = Node_state.get_initial_state ~folder:node_folder ~minimum_block_delay:0. ~pollinate_context:Node.State.Client in 
   let payload =
-    let open Trusted_validators_membership_change in
     { address; action = Add } in
   let payload_json_str =
     payload
-    |> Trusted_validators_membership_change.payload_to_yojson
+    |> validators_payload_to_yojson
     |> Yojson.Safe.to_string in
+
   let payload_hash = BLAKE2B.hash payload_json_str in
   let signature = Signature.sign ~key:identity.secret payload_hash in
-  let%await () =
-    Network.request_trusted_validator_membership { signature; payload }
-      identity.uri in
+  let%await interop_context = interop_context node_folder in
+  let%await validator_uris = validator_uris ~interop_context in
```

Comment:
Fetching validator_uris through Tezos is actually pretty slow when we wait for several confirmations, so you should not add this operation when it is not needed. Here you're modifying code that wasn't doing that (I believe it actually was removed fairly recently), because it's not needed.